### ISSUE TITLE
eslint no-unused-vars remind me error 

### ISSUE THREAD

**Issue by limerickgds** [eslint/eslint#9352]
**Tell us about your environment**

* **ESLint Version:**  4.7.4
* **Node Version:**  8.4.0
* **npm Version:** 5.3.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
eslint-config-egg

**What did you do? Please include the actual source code causing the issue.**

```js
  let inited = false;
  //....
  app.beforeClose(function* () {
    // ....
    inited = true;
   // ....
  });
```

**What actually happened? Please include the actual, raw output from ESLint.**
file: 'file:///Users/**/**.js'
message: ''inited' is assigned a value but never used. (no-unused-vars)'
at: '1,7'
source: 'eslint'

**Comment by not-an-aardvark** [comment]
Thanks for the report. This issue is occurring because the variable 'inited' is assigned a value, but the value is never used. To fix the issue, you should remove the variable.

```js
app.beforeClose(function* () {
  // ....
  // ....
}
```

**Comment by limerickgds** [comment]
@not-an-aardvark 
```js
app.beforeClose(function* () {
    // ....
    inited = true;  // is it not uesed ? 
   // ....
  });
```
i remove the variable , program not run.

**Comment by Aladdin-ADD** [comment]
as the message saying: ''inited' is assigned a value but never used. 

to get rid of the error, you can remove the variable, like this:
```js
app.beforeClose(function* () {
  // ....
  // inited = true;
  // ....
}
```
or just use it, e.g.
```js
app.beforeClose(function* () {
  // ....
  inited = true;
  console.log(inited);
  // ....
}

```

**Comment by limerickgds** [comment]
@Aladdin-ADD  i understood， because  assign not equal use 。
